algo/src/test/java/com/hartwig/actin/algo/evaluation/general/HasMaximumBMITest.java

```diff
@@ -0,0 +1,67 @@
+package com.hartwig.actin.algo.evaluation.general;
+
+import com.hartwig.actin.algo.datamodel.Evaluation;
+import com.hartwig.actin.algo.datamodel.EvaluationResult;
+import com.hartwig.actin.clinical.datamodel.ImmutableBodyWeight;
+import org.junit.Test;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static com.hartwig.actin.algo.evaluation.EvaluationAssert.assertEvaluation;
+import static org.junit.Assert.assertTrue;
+
+public class HasMaximumBMITest {
+    private final HasMaximumBMI function = new HasMaximumBMI(40);
+    private final LocalDate now = LocalDate.now();
+    private final LocalDate lastYear = now.minusYears(1);
+
+    @Test
+    public void shouldBeUndeterminedWhenNoBodyWeightsProvided() {
+        assertEvaluation(EvaluationResult.UNDETERMINED,
+                function.evaluate(GeneralTestFactory.withBodyWeights(Collections.emptyList())));
+    }
+
+    @Test
+    public void shouldBeUndeterminedWhenNoBodyWeightsProvidedWithExpectedUnit() {
+        assertEvaluation(EvaluationResult.UNDETERMINED, function.evaluate(GeneralTestFactory.withBodyWeights(
+                Collections.singleton(
+                        ImmutableBodyWeight.builder().date(now).value(70).unit("pound").build()
+                )
+        )));
+    }
+
+    @Test
+    public void shouldPassIfLatestWeightIsLessThanWarnThreshold() {
+        Evaluation evaluation = function.evaluate(GeneralTestFactory.withBodyWeights(Arrays.asList(
+                ImmutableBodyWeight.builder().date(now).value(70.57).unit("Kilogram").build(),
+                ImmutableBodyWeight.builder().date(lastYear).value(100.0).unit("Kilogram").build()
+        )));
+        assertEvaluation(EvaluationResult.PASS, evaluation);
+        assertTrue(evaluation.passSpecificMessages().contains(
```

Comment:
I have never tested the exact messages, feels that adds too little for the overhead it causes. 

The assertEvaluation function checks if the expected message types are provided, but not the actual content. 

Do you think we should test for the actual message content?

Reply:
I included the messages in testing because they're composed in the code and I wanted to verify the final product. We can drop the tests if they're too brittle.

Reply:
Locale is fixed in #4.